src/onnx/parse_if.cpp

```diff
@@ -65,13 +65,39 @@ struct parse_if : op_parser<parse_if>
 
         auto then_out_shapes = then_mdl->get_output_shapes();
         auto else_out_shapes = else_mdl->get_output_shapes();
-        if(not std::equal(then_out_shapes.begin(),
-                          then_out_shapes.end(),
-                          else_out_shapes.begin(),
-                          else_out_shapes.end()))
+
+        assert(then_out_shapes.size() == else_out_shapes.size());
+
+        // Must have the same type for both if/else blocks by onnx spec
+        if(then_out_shapes.at(0).type() != else_out_shapes.at(0).type())
         {
             MIGRAPHX_THROW("PARSE_IF: " + info.name +
-                           " then and else sub_grahps must have same output shapes!");
+                           " then and else sub_grahps must have same output type! " +
+                           std::to_string(then_out_shapes.at(0).type()) + " vs " +
+                           std::to_string(else_out_shapes.at(0).type()));
+        }
+
+        // Need to check static shapes result
+        if(not then_out_shapes.at(0).dynamic() && not else_out_shapes.at(0).dynamic())
+        {
+            if(then_out_shapes.at(0).scalar())
+            {
+                if(then_out_shapes.at(0).lens().at(0) != else_out_shapes.at(0).lens().at(0) ||
+                   then_out_shapes.at(0).strides().at(0) != 1)
```

Comment:
MIGraphX definition of scalar is that `strides` should be zero. 
Also if it is scalar then, it is not dynamic.

Reply:
`then_out_shapes.at(0).strides().at(0) != 1` is always true if it is a scalar.

Reply:
Fixed